js/app/demo/app.jsx

```diff
@@ -15,7 +15,7 @@ define(['react', 'jsx!editor', 'jsx!messages', 'jsx!fixedCode', 'jsx!configurati
                 return result;
             }
 
-            return {
+            var initialState = JSON.parse(window.localStorage.getItem('linterDemoState')) || {
```

Comment:
Good point. I think merging state would be better -- it seems annoying to invalidate it every time a new version comes out. (Personally I would want to set a configuration once and then never have to deal with the default configuration again.)